asio/include/asio/detail/recycling_allocator.hpp

```diff
@@ -29,6 +29,8 @@ template <typename T>
 class recycling_allocator
 {
 public:
+  typedef T value_type;
+
   template <typename U>
   struct rebind
```

Comment:
This `rebind` template (and the one in `recycling_allocator<void>`) could even be removed, because `allocator_traits` will do the right thing even without it.